### PR TITLE
Add telemetry test extension to manifest.

### PR DESCRIPTION
**Pull request by jrbenny35**
https://bugzilla.mozilla.org/show_bug.cgi?id=1556057

### CODE REVIEW

**Review by escapewindow — COMMENTED** [review]
Thank you!

I see the repo doesn't currently have [branch protections](https://github.com/mozilla-extensions/dynamic-probe-telemetry-extension/settings/branches) set. Would we be able to do that? (This would change your workflow to require review before merging PRs; do you have someone else who can review your changes?)

File: xpi-manifest.yml [hunk 1]
```diff
@@ -193,4 +193,13 @@ xpis:
...
     addon-type: privileged
+    install-type: npm
+  - name: telemetry-testing-extension
+    description: Extension for testing telemetry APIs
+    repo-prefix: dynamic-probe-telemetry-extension
+    active: true
+    private-repo: false
+    artifacts:
+      - web-ext-artifacts/dynamic-probe-telemetry-extension.xpi
+    addon-type: privileged
```
[the remark this answers is not in the document]
Comment (jrbenny35) [reply xpi-manifest.yml:204]: No, it is purely for testing the `telemetry` webext API. See: https://bugzilla.mozilla.org/show_bug.cgi?id=1556057

[the remark this answers is not in the document]
Comment (jrbenny35) [reply xpi-manifest.yml:204]: Also, I don't think I can change the branch protections list myself. If you can do that, that would be great 😄

[the remark this answers is not in the document]
Comment (escapewindow) [reply xpi-manifest.yml:204]: I don't know where my other comment went, but I noticed your CI dep signing is getting the error `
signingscript.exceptions.SigningScriptError: /app/workdir/public/build/dynamic-probe-telemetry-extension.xpi is not a valid xpi`. This probably means you don't have `applications.gecko.id` defined in your manifest. (You may have `browser_specific_settings` instead, which is [not yet supported](https://github.com/mozilla-releng/scriptworker-scripts/issues/221).

[the remark this answers is not in the document]
Comment (jrbenny35) [reply xpi-manifest.yml:204]: Just updated the repo

**Review by escapewindow — APPROVED** [review]
Thanks!

**Pull request merged**